### PR TITLE
Fix: 500 error on Dataset validation

### PR DESCRIPTION
**Pull request by Geeker1**
## Description
During dataset upload, an Attribute error is thrown if profile is None, fixed this by writing the clean method for that field,  raising a Validation error if profile is None.

## Related Issue
#204

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Review by milafrerichs — CHANGES_REQUESTED** [review]
this not the intended solution.

the profile is not mandatory. but the profile is mandatory when the permission type is private. so we need a check here but only if the permission is private. 

What we need to do is to check the code here:

https://github.com/OpenUpSA/wazimap-ng/blob/staging/wazimap_ng/datasets/admin/dataset_admin.py#L113-L114

And only run that if the profile is set.

File: wazimap_ng/datasets/admin/forms/dataset_admin_form.py [hunk 1]
```diff
@@ -6,6 +6,17 @@
 class DatasetAdminForm(forms.ModelForm):
     import_dataset = forms.FileField(required=False)
 
+    def clean(self):
+        cleaned_data = super(DatasetAdminForm, self).clean()
+
+        profile = cleaned_data.get('profile', None)
+        permission_type = cleaned_data.get('permission_type', None)
+
+        if not profile:
```
Comment (milafrerichs) [inline wazimap_ng/datasets/admin/forms/dataset_admin_form.py:15]: I would merge those together and make it more obvious
`if permission_type == 'private' and profile == None`

**Review by milafrerichs — APPROVED** [review]
